.azure/bicep/main.bicep

```diff
@@ -1,35 +1,202 @@
-param env string = 'dev'
+// This deploys an entire environment stack*
+// It reuses some shared resources within a resource group (e.g. prod / non-prod)
+// and then deploys and configures environment specific resources
+// based on parameters passed through
+// for a given service and environment combination (e.g. decsys dev)
 
-param keyVaultName string
-param appName string
-param logAnalyticsWorkspaceName string
+import { referenceSecret } from 'br/DrsUtils:functions:v1'
+import { ConnectionStringDictionary } from 'br/DrsUtils:types:v1'
 
-resource kv 'Microsoft.KeyVault/vaults@2020-04-01-preview' existing = {
+type ServiceNames = 'decsys'
+param serviceName ServiceNames
+
+type Environments = 'dev' | 'qa' | 'uat' | 'prod'
+param env Environments
+
+param frontendAppName string = '${env}-${serviceName}'
+param frontendHostnames array = []
+
+param backendAppName string = '${env}-${serviceName}-api'
+param appSettings object = {}
+
+param keyVaultName string = '${serviceName}-${env}-kv'
+param location string = resourceGroup().location
+
+param sharedEnv string = 'shared'
+var sharedPrefix = serviceName
+param appServicePlanSku string = 'B2'
+
+// log analytics workspace
+module la 'br/DrsComponents:log-analytics-workspace:v1' = {
+  name: 'la-ws-${uniqueString(sharedPrefix)}'
+  params: {
+    location: location
+    logAnalyticsWorkspaceName: '${sharedPrefix}-la-ws'
+    tags: {
+      ServiceScope: serviceName
+      Environment: sharedEnv
+    }
+  }
+}
+
+// App Service Plan
+module asp 'br/DrsComponents:app-service-plan:v1' = {
+  name: 'asp'
+  params: {
+    location: location
+    aspName: '${sharedPrefix}-asp'
+    sku: appServicePlanSku
+    tags: {
+      ServiceScope: serviceName
+      Environment: sharedEnv
+    }
+  }
+}
+
+// Per Environment Resources
+
+// Environment Key Vault pre-existing and populated
+resource kv 'Microsoft.KeyVault/vaults@2019-09-01' existing = {
   name: keyVaultName
 }
 
-module asp './components/app-service-plan.bicep' = {
-  name: 'asp-${uniqueString(env)}'
+// Create the Frontend App and related bits
+// App Insights
+// App Service
+// Hostnames
+module frontend 'br/DrsComponents:app-service:v1' = {
```

Comment:
rename `frontend` to just `decsys`, or `app` or something, since we don't have separate front and backend apps in decsys today.